components/filters.tsx

```diff
@@ -57,39 +59,77 @@ export const Filters = () => {
     );
   };
 
+  function loadFilters(data: { [x: string]: any; }) {
+    // set defaults or load from arg
+    const setData = {
+      min_sa: data["min_sa"] ?? 0,
+      max_sa: data["max_sa"] ?? 99,
+      min_drivers: data["min_cd"] ?? 0,
+      max_drivers: data["max_cd"] ?? 99,
+      min_tm: data["min_tm"] ?? 0,
+      max_tm: data["max_tm"] ?? 3,
+      showEmpty: data["showEmpty"] ?? false ? true : false,
+      showFull: data["showFull"] ?? true ? true : false,
+      class: data["class"] ?? ["mixed", "gt3", "gt4", "gtc", "tcx", "gt2"],
+      dlc: data["dlc"] ?? [
+        "base",
+        "icgt",
+        "gtwc",
+        "bgt",
+        "atp",
+        "gt2",
+        "nurb-24",
+      ],
+      track: data["track"] ?? Object.keys(staticData.tracks),
+      sessions: data["session"] ?? ["race", "qualifying", "practice"],
+      favouritesOnly: data["favouritesOnly"] ?? false ? true : false,
+    };
+    setFilterData(setData);
+  }
+
   useEffect(() => {
     if (filterData) {
       localStorage.setItem("filters", JSON.stringify(filterData));
       ctx.refetch(filterData);
     } else {
-      // set defaults or load from localStorage
       const data = JSON.parse(localStorage.getItem("filters") ?? "{}");
-      const setData = {
-        min_sa: data["min_sa"] ?? 0,
-        max_sa: data["max_sa"] ?? 99,
-        min_drivers: data["min_cd"] ?? 0,
-        max_drivers: data["max_cd"] ?? 99,
-        min_tm: data["min_tm"] ?? 0,
-        max_tm: data["max_tm"] ?? 3,
-        showEmpty: data["showEmpty"] ?? false ? true : false,
-        showFull: data["showFull"] ?? true ? true : false,
-        class: data["class"] ?? ["mixed", "gt3", "gt4", "gtc", "tcx", "gt2"],
-        dlc: data["dlc"] ?? [
-          "base",
-          "icgt",
-          "gtwc",
-          "bgt",
-          "atp",
-          "gt2",
-          "nurb-24",
-        ],
-        sessions: data["session"] ?? ["race", "qualifying", "practice"],
-        favouritesOnly: data["favouritesOnly"] ?? false ? true : false,
-      };
-      setFilterData(setData);
+      loadFilters(data);
     }
   }, [filterData]);
 
+  const selectAllTracks = () => {
+    let tracks = Object.keys(staticData.tracks);
+    localStorage.setItem("filters", JSON.stringify(tracks))
+    loadFilters({ track: tracks})
+  }
+
+  const selectNoneTracks = () => {
+    localStorage.setItem("filters", "{}")
+    loadFilters({ track: [] })
+  }
+
+  const trackFilters = [
+    <span key="track.title" className={styles.filter_group_name}>Tracks
+      <span key="track.all" className={styles.filter_button} onClick={selectAllTracks}>ALL</span>
+      <span key="track.none" className={styles.filter_button} onClick={selectNoneTracks}>NONE</span>
+    </span>,
+    ...Object.keys(staticData.tracks).map(id => {
+      const track = {id,...staticData.tracks[id]};
```

Comment:
So I'm not that worried about styling however this list may be too large that you cannot close the filter menu on mobile for example. I don't mind doing this part myself but if you fancy a challenge, feel free to try and make it scrollable :)